components/typography/metadata/typography.yml

```diff
@@ -35,11 +35,77 @@ examples:
     markup: |
       <p class="spectrum-Detail spectrum-Detail--XL">Our recommendations</p>
   - id: code-m
-    name: Typography - Code
+    name: Code
     status: Verified
     description: |
       Code is used for text that represents code.
 
       [View the Code typography component](typography-code.html)
     markup: |
       <code class="spectrum-Code spectrum-Code--S">alert("Hello world");</code>
+  - id: migrating-typography
+    status: Verified
+    name: Migrating from deprecated Typography
+    description: |
+      See the table below to reference what will need to change when migrating to the new Typography API. Cells with a dash indicate the typography style did not exist in the previous API.
+
+      Note that all instances of typography now require the component class in addition to the modifier class. For example, to get a large Heading, you will need the `.spectrum-Heading--L` modifier class as well as the `.spectrum-Heading` component class.
+
+      #### Heading
+
+      | Deprecated Classname                                    | New Classname                                                      |
+      | ------------------------------------------------------- | ------------------------------------------------------------------ |
+      | `~.spectrum-Heading1--display`                           | `.spectrum-Heading.spectrum-Heading--XXXL`                         |
```

Comment:
Note the `~` at the start of the inline code block -- I extended `markdown-it` to add a class to turn this code red per @NateBaldwinDesign

Reply:
I personally would have solved this by id'ing the markdown and add the styling via CSS. Your approach requires to loop over the content of the markdown.
I heard this is a way that works for some teams: https://lifelongprogrammer.blogspot.com/2019/01/how-to-style-markdown-with-css.html

Reply:
Unfortunately, that syntax doesn't seem to be supported: 
![image](https://user-images.githubusercontent.com/201344/74066964-0200d400-49ad-11ea-89f2-58f99c6252b0.png)

It does seem we could use [custom containers](https://github.com/markdown-it/markdown-it-container) and apply `::: spectrum-CSSComponent-migrationTable` around the table, then style `.spectrum-Table-body .spectrum-Table-cell:first-child` to accomplish this, but that's quite a bit less direct than the (admittedly weird) approach taken. Thoughts?

Reply:
Adding a class name on custom containers is what I would do and not id-ing each element
would this work around the content placed in a the table?

```
::: warning
*here be dragons*
:::
```
to hopefully:
```
<div class="warning">
<em>here be dragons</em>
</div>
```
and then we could target the 'warning' class by renaming it and styling it in CSS?

Another alternative could be to use the sibling selector to style the content followed after a selector.  https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator

This personally feels to me a bit more direct than looping over a utility function over the content. But that's just my own personal feeling and I am open for other suggestions.

Reply:
Yeah we could wrap every single instance of the code block with the `:::warning` bit, that would do it. I'll make that change.

Reply:
@bernhard-adobe uhg, that doesn't work. Because of the way tables are formatted, we can't put a block element inside of it. I can fall back to the outer class and use the `:first-child` bit, but to be honest, that's not really gaining us anything. 

The tokenization function runs already, we're just replacing the rule with our own, so we're not adding any runtime complexity or slowing things down (docs generation takes 5 seconds).

Would you block this pull request pending a different solution, or are you ok with the existing solution?

Reply:
Thank you @lazd  for your customization.
It is still unclear to me why you would prefer writing those rules in JavaScript as an utility function if you could use a wrapper / container ? https://github.com/markdown-it/markdown-it-container This feels more like a straight forward solution to me that might help in the future with area in the documentation that could benefit from customization.

Reply:
So currently, we use markdown in our docs. When we write the following:

```
      Heading is used to create various levels of typographic hierarchies.
      [View the Heading typography component](typography-heading.html)
```

It automatically turns into this:

```
<p>Heading is used to create various levels of typographic hierarchies.</p>
<p><a href="typography-heading.html">View the Heading typography component</a></p>
```

That looks like the following:
![image](https://user-images.githubusercontent.com/201344/74077076-a1d05900-49d1-11ea-9339-ea32d48b07e6.png)

You can see it looks wrong because we're missing the `spectrum-Link` class, right?

So, I've added rules to the markdown parser such that it will output the following HTML:

```
<p>Heading is used to create various levels of typographic hierarchies.</p>
<p><a href="typography-heading.html" class="spectrum-Link">View the Heading typography component</a></p>
```

That looks like the following:

![image](https://user-images.githubusercontent.com/201344/74077095-bdd3fa80-49d1-11ea-8c9e-7eabdcd7952c.png)

Now you can see, it's a real Spectrum link! 

Of course, we could write HTML inside of our markdown and get the same result:

```
      Heading is used to create various levels of typographic hierarchies.
      <a href="typography-heading.html">View the Heading typography component</a>
```

But why would we want to do that? Instead, our markdown is automatically rendered with the correct Spectrum CSS classes.

I did the same for `### Headings`, ``code``, and tables. This is awesome, right?

In the case of the red code blocks, however, I extended the existing code block to add another class if it finds a `~` as the first char. Other approaches with the wrapper were not feasible, and wrapping the tables and using overly specific CSS to target it doesn't make sense.

It's magical and super awesome, and all of the potential solutions either don't work or are tedious and silly. We're using `markdown-it` precisely how it was designed, and there are tons of issues filed against their repository asking how to do exactly what we're doing, documentation that shows how to do exactly what we're doing etc.